db/scripts/find_cured.py

```diff
@@ -1,100 +1,105 @@
+    cursor.execute(queries.get_rejected_NC(election))
+    output = cursor.fetchall()
 
-  # for each rejected entry, add to rejected table
-  for entry in output:
-    cursor.execute(queries.add_to_rejected_NC(rejected_db, entry))
-    mydb.commit()
+    # for each rejected entry, add to rejected table
+    for entry in output:
+      cursor.execute(schema.add_to_rejected_NC(rejected_db, entry))
 
+  mydb.commit()
   # To close the connection
   mydb.close()
 
 
 # Adds entries to cured and rejected tables
 def find_cured(today_datetime, state):
-  table = config[state]['table']
-  rejected_db = "rejected_"+table
-  cured_db = "cured_"+table
   mydb = MySQLdb.connect(host=config['DATABASE']['host'],
               user=config['DATABASE']['user'],
               passwd=config['DATABASE']['passwd'],
               db=config[state]['db'],
               local_infile=1)
   print("Connected to db")
 
-  cursor = mydb.cursor(MySQLdb.cursors.DictCursor)
+  cursor = mydb.cursor()
+
+  elections = get_elections(cursor)
+  
+  for election in elections:
+  
+    cured_db = f'cured_{election}'
+    rejected_db = f'rejected_{election}'
 
-  # make cured table if not made
-  cursor.execute(queries.create_cured_table(cured_db))
+    # make cured table if not made
+    cursor.execute(schema_table(cured_db))
 
-  # make rejected table if not made
-  cursor.execute(queries.create_rejected_table(rejected_db))
+    # make rejected table if not made
+    cursor.execute(schema_table(rejected_db))
   
-  # get all cured ballots as of today
-  print("Get all cured ballots")
-  cursor.execute(queries.get_cured(table))
-  output = cursor.fetchall()
-
-  # for each cured entry, add to cured_db
-  for entry in output:
-    cursor.execute(queries.add_to_cured(cured_db, entry))
-    mydb.commit()
-
-  # query the current day for any new rejected that are not cured
-  print("Getting today's rejected ballots from main table")
-  cursor.execute(queries.get_today_rejected(table, today_datetime, cured_db))
-  output = cursor.fetchall()
-
-  # for each rejected entry today, add to rejected table
-  for entry in output:
-    cursor.execute(queries.add_to_rejected(
-      rejected_db, entry))
-    mydb.commit()
+    # get all cured ballots as of today
+    print("Get all cured ballots")
+    cursor.execute(queries.get_cured(election))
+    output = cursor.fetchall()
+
+    # for each cured entry, add to cured_db
+    for entry in output:
+      cursor.execute(schema.add_to_cured(cured_db, entry))
+
+    # query the current day for any new rejected that are not cured
+    print("Getting today's rejected ballots from main table")
+    cursor.execute(queries.get_today_rejected(election, today_datetime, cured_db))
+    output = cursor.fetchall()
+
+    # for each rejected entry today, add to rejected table
+    for entry in output:
+      cursor.execute(schema.add_to_rejected(rejected_db, entry))
 
   # To close the connection
   mydb.close()
```

Comment:
I saw you deleted the incremental "commit()" in both this method and the find_cured_NC() method. In the NC method, you added a "commit()" right before closing the connection but there isn't one here. Should there be or is that done some other way I'm missing?

Reply:
I moved the incremental commit since it was committing changes to the db on every single row, which is definitely not optimal. I moved it to commit the changes after all the rows are processed. And good catch - I forgot to put the last commit back, thanks!